internal/pkg/earpointer/analysis_test.go

```diff
@@ -348,10 +439,20 @@ func TestStore(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	want :=
-		`{*f.a}: [x->f.t0, y->f.t2], {f.a}: --> *f.a, {f.b}: [10->f.t3], {f.t0}: --> f.t1, {f.t1}: [], {f.t2}: --> f.t4, {f.t3}: --> f.t4, {f.t4}: [], {z}: --> f.t1`
-	if got := state.String(); got != want {
-		t.Errorf("got: %s\n want: %s", got, want)
+	want := map[string]string{
+		"{f.a}":  "--> *f.a",
+		"{*f.a}": "[x->f.t0, y->f.t2]",
+		"{f.b}":  "[10->f.t3]",
+		"{f.t0}": "--> f.t1",
+		"{f.t1}": "[]",
+		"{f.t2}": "--> f.t4",
+		"{f.t3}": "--> f.t4",
+		"{f.t4}": "[]",
+		"{z}":    "--> f.t1",
+	}
+	wantstr := partitionItemsToString(want)
```

Comment:
All of these tests roughly have the same pattern. Seems like a good candidate for a table based test. Zvonimir had a decent setup you might want to take a look at. Used multiple files though.

Reply:
Yes writing the expected string into a text file or with some format will be surely more readable. But it requires a reader to open the file to compare. I think that a better one may be write the state as a Diagonistics and then everything is in the same source file. However vet's Diagonistics is basically one line so it may not be very readable if the state is large.
I am OK with the map-based solution for the moment. More cleaning up can be done in the future.